homeassistant/components/homeassistant/scene.py

```diff
@@ -271,6 +282,28 @@ async def create_service(call: ServiceCall) -> None:
         SCENE_DOMAIN, SERVICE_CREATE, create_service, CREATE_SCENE_SCHEMA
     )
 
+    async def delete_service(call: ServiceCall) -> None:
+        """Delete a dynamically created scene."""
+        entity_id = call.data[CONF_ENTITY_ID]
+        scene = platform.entities.get(entity_id)
+        if scene is None:
+            _LOGGER.error("The scene %s does not exist", entity_id)
+            return
+        if not isinstance(scene, HomeAssistantScene):
+            _LOGGER.error("Entity %s is not a scene", entity_id)
+            return
+        if not scene.from_service:
+            _LOGGER.error(
+                "The scene %s is not created with service `scene.create`", entity_id
+            )
+            return
+
+        await platform.async_remove_entity(entity_id)
+
+    hass.services.async_register(
```

Comment:
This should probably be an entity service as it targets an entity. For those kinds of services we have a helper for registration in the entity platform.

```py
platform = entity_platform.async_get_current_platform()
platform.async_register_entity_service(
    ...
)
```